### ISSUE TITLE
Choose hostname when `cargo shuttle run`

### ISSUE THREAD

**Issue by tguichaoua** [shuttle-hq/shuttle#350]
It could be usefull when debuging to host the server not only on the `localhost` but also on network to allow other devices to send request.
For example [`vite`](https://vitejs.dev/) has a `--host` flag for this.

```
  VITE v3.1.3  ready in 399 ms

  ➜  Local:   http://localhost:5173/
  ➜  Network: http://192.168.1.100:5173/
```

**Comment by trkohler** [comment]
hello! can i take this one as a part of hacktoberfest?

**Comment by chesedo** [comment]
@0xpwnstar feel free to take it if you are still interested

**Comment by joshua-mo-143** [comment]
I've been working on this over the past few days and believe I may have a solution to this problem - just need to do some testing and I will hopefully be able to push a PR if all goes well.

**Comment by chesedo** [comment]
@joshua-mo-143 sounds exciting! :partying_face:

**Comment by chesedo** [comment]
This is closed by #565
